feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/data/repository/WalletRepositoryImpl.kt

```diff
@@ -173,6 +173,9 @@ class WalletRepositoryImpl(
         phishingAddresses.contains(addressPublicKey)
     }
 
+    override suspend fun getAccountFreeBalance(accountAddress: String) =
+        substrateSource.getAccountInfo(accountAddress).data.free.toBigDecimal()
```

Comment:
I suggest not to use BigDecimal here since
1. The data layer (aka repositories) currently returns balances in planks (BigInteger) and then the domain or presentation layer converts it to a readable amount with Token instance (BigDecimal)
2. From the previous point you can see, that we have a kinda implicit agreement that BigInteger -> planks, BigDecimal -> 'real' amount

Reply:
Done